### PR TITLE
Bug 1723130 - List necessary webext permissons for using Glean.js [doc only]

### PR DESCRIPTION
**Pull request by brizental**

### CODE REVIEW

File: docs/user/user/adding-glean-to-your-project/javascript.md [hunk 1]
```diff
@@ -8,15 +8,19 @@ Currently, these bindings support collecting data from [Browser Extensions](http
 
 ## Requirements
 
-* Node.js >= 12.20.0.
-* npm >= 7.0.0.
-* Webpack >= 5.34.0.
-* Python >= 3.6.[^1]
-* (_for browsers other than Firefox_) [webextension-polyfill](https://github.com/mozilla/webextension-polyfill) >= 0.8.0.[^2]
+* Node.js >= 12.20.0;
```
Comment (badboy) [inline docs/user/user/adding-glean-to-your-project/javascript.md:11]: huh, feels weird to see a `;` here. Can we just leave them out (no dot either)?

File: docs/user/user/adding-glean-to-your-project/javascript.md [hunk 2]
```diff
@@ -8,15 +8,19 @@ Currently, these bindings support collecting data from [Browser Extensions](http
...
+* npm >= 7.0.0;
+* Webpack >= 5.34.0;
+* Python >= 3.6.
+  * The `glean` command requires Python to download [`glean_parser`](https://mozilla.github.io/glean_parser/) which is a Python library.
 
-[^1]: The `glean` command requires Python to download [`glean_parser`](https://mozilla.github.io/glean_parser/) which is a Python library.
+### Web extension specific requirements
 
-[^2]: Glean.js assumes a Promise-based `browser` API: Firefox provides such an API by default. Other browsers may require using a polyfill library such us `webextension-polyfill` when using Glean in browser extensions.
+* ["storage"](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json/permissions#api_permissions) API permission;
```
Comment (badboy) [inline docs/user/user/adding-glean-to-your-project/javascript.md:19]: Maybe an example how that is set?

File: docs/user/user/adding-glean-to-your-project/javascript.md [hunk 3]
```diff
@@ -8,15 +8,19 @@ Currently, these bindings support collecting data from [Browser Extensions](http
...
+* [Host permission](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json/permissions#host_permissions) to the Telemetry server;
+  * Only necessary if the defined server endpoint denies [cross-origin](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS) requests.
```
Comment (badboy) [inline docs/user/user/adding-glean-to-your-project/javascript.md:21]: So it's not necessary for our endpoint? If so please call that out.

File: docs/user/user/adding-glean-to-your-project/javascript.md (hunk 2 above)
Comment (badboy) [reply docs/user/user/adding-glean-to-your-project/javascript.md:19]: (no `;` here either)

File: docs/user/user/adding-glean-to-your-project/javascript.md [hunk 4]
```diff
@@ -8,15 +8,19 @@ Currently, these bindings support collecting data from [Browser Extensions](http
...
+* (_for browsers other than Firefox_) [webextension-polyfill](https://github.com/mozilla/webextension-polyfill) >= 0.8.0.[^1]
```
Comment (badboy) [inline docs/user/user/adding-glean-to-your-project/javascript.md:22]: This list has several different things. Please call out that this is another npm dependency for the project

**Review by badboy — CHANGES_REQUESTED** [review]

File: docs/user/user/adding-glean-to-your-project/javascript.md (hunk 3 above)
Comment (brizental) [reply docs/user/user/adding-glean-to-your-project/javascript.md:21]: It is necessary for our endpoint.

Comment (badboy) [reply docs/user/user/adding-glean-to-your-project/javascript.md:21]: Then maybe making this more explicit would be good

File: docs/user/user/adding-glean-to-your-project/javascript.md (hunk 4 above)
Comment (brizental) [reply docs/user/user/adding-glean-to-your-project/javascript.md]: I tried to add some txt to clarify this, but anything I wrote sounded confusing. It's not really a npm dependency, you can just download the script and load it. The important part is that you load it... Anyways, I added a link to the sample configuration, that is probably much more enlightening than anything I write.

**Review by badboy — APPROVED** [review]

File: docs/user/user/adding-glean-to-your-project/javascript.md (hunk 4 above)
Comment (badboy) [reply docs/user/user/adding-glean-to-your-project/javascript.md]: ah well, yeah, some sort of source dependency for users.

**Pull request merged**